webui/react/src/pages/ModelRegistry.tsx

```diff
@@ -468,17 +459,21 @@ const ModelRegistry: React.FC<Props> = ({ workspace }: Props) => {
       </div>
     );
 
-    const actionRenderer = (_: string, record: ModelItem) => (
-      <Dropdown menu={ModelActionMenu(record)} trigger={['click']}>
-        <Button icon={<Icon name="overflow-vertical" />} type="text" />
-      </Dropdown>
-    );
+    const actionRenderer = (_: string, record: ModelItem) => {
+      const canDelete = canDeleteModel({ model: record });
+      const canModify = canModifyModel({ model: record });
+      return canModify || canDelete ? (
+        <Dropdown menu={ModelActionMenu(record, canDelete, canModify)} trigger={['click']}>
+          <Button icon={<Icon name="overflow-vertical" />} type="text" />
+        </Dropdown>
+      ) : null;
+    };
 
     const descriptionRenderer = (value: string, record: ModelItem) => (
       <Input
         className={css.descriptionRenderer}
         defaultValue={value}
-        disabled={record.archived}
+        disabled={record.archived || !canModifyModel({ model: record })}
```

Comment:
i think we dont need to call `canModifyModel` func here, right? cuz `canModify` (`canModifyModelFlag`) is defined above

Reply:
At this part of ModelRegistry we are inside the cell renderer for each model-row. `const columns = useMemo(() => { ... const descriptionRenderer = (value: string, record: ModelItem) => ( ... )... }`

In ModelRegistry canModifyModelFlag is set only in the three-dots menu and right-click / action menu